buildspec.yml

```diff
@@ -9,12 +9,25 @@ phases:
         - npm install -g yarn
     build:
       commands:
-        - mkdir -p fhir-works-on-aws-interface && cp -r $CODEBUILD_SRC_DIR_FhirWorksInterface/* ./fhir-works-on-aws-interface && mv fhir-works-on-aws-interface ../
-        - (cd ../fhir-works-on-aws-interface && yarn install)
-        - mkdir -p fhir-works-on-aws-routing && cp -r $CODEBUILD_SRC_DIR_FhirWorksRouting/* ./fhir-works-on-aws-routing && mv fhir-works-on-aws-routing ../
-        - (cd ../fhir-works-on-aws-routing && yarn install)
-        - mkdir -p fhir-works-on-aws-search-es && cp -r $CODEBUILD_SRC_DIR_FhirWorksSearchES/* ./fhir-works-on-aws-search-es && mv fhir-works-on-aws-search-es ../
-        - (cd ../fhir-works-on-aws-search-es && yarn install)
-        - mkdir -p fhir-works-on-aws-persistence-ddb && cp -r $CODEBUILD_SRC_DIR_FhirWorksPersistenceDDB/* ./fhir-works-on-aws-persistence-ddb && mv fhir-works-on-aws-persistence-ddb ../
-        - (cd ../fhir-works-on-aws-persistence-ddb && yarn install)
-        - yarn install
\ No newline at end of file
+        # AWS doesn't like symlinks, so we have to go around this
+        - PACKAGE=../fhirworks
+
+        - mkdir $PACKAGE
+        - mkdir -p $PACKAGE/fhir-works-on-aws-deployment && cp -r . $PACKAGE/fhir-works-on-aws-deployment
+
+        - mkdir -p $PACKAGE/fhir-works-on-aws-interface && cp -r $CODEBUILD_SRC_DIR_FWInt/* $PACKAGE/fhir-works-on-aws-interface
+        - (cd $PACKAGE/fhir-works-on-aws-interface && yarn install)
+        - mkdir -p $PACKAGE/fhir-works-on-aws-routing && cp -r $CODEBUILD_SRC_DIR_FWRout/* $PACKAGE/fhir-works-on-aws-routing
+        - (cd $PACKAGE/fhir-works-on-aws-routing && yarn install)
+        - mkdir -p $PACKAGE/fhir-works-on-aws-search-es && cp -r $CODEBUILD_SRC_DIR_FWSES/* $PACKAGE/fhir-works-on-aws-search-es
+        - (cd $PACKAGE/fhir-works-on-aws-search-es && yarn install)
+        - mkdir -p $PACKAGE/fhir-works-on-aws-persistence-ddb && cp -r $CODEBUILD_SRC_DIR_FWPDDB/* $PACKAGE/fhir-works-on-aws-persistence-ddb
+        - (cd $PACKAGE/fhir-works-on-aws-persistence-ddb && yarn install)
+
+        - (cd $PACKAGE/fhir-works-on-aws-deployment && yarn install)
```

Comment:
Why copy everything? Why not do everything in place and then add to archive what's needed?

Reply:
That's because the dependencies on npm are configured as local folder dependencies instead of git packages. In order for the yarn/npm install to work, the folders have to exist in the relative path, and then npm creates symbolic links in the node_modules for that folders. https://docs.npmjs.com/cli/v7/configuring-npm/package-json#local-paths